go-controller/pkg/ovn/controller/egress_services/egress_services_controller.go

```diff
@@ -410,3 +455,316 @@ func (c *Controller) repair() error {
 
 	return nil
 }
+
+// onEgressServiceAdd queues the EgressService for processing.
+func (c *Controller) onEgressServiceAdd(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %+v: %v", obj, err))
+		return
+	}
+	c.egressServiceQueue.Add(key)
+}
+
+// onEgressServiceUpdate queues the EgressService for processing.
+func (c *Controller) onEgressServiceUpdate(oldObj, newObj interface{}) {
+	oldEQ := oldObj.(*egressserviceapi.EgressService)
+	newEQ := newObj.(*egressserviceapi.EgressService)
+
+	if oldEQ.ResourceVersion == newEQ.ResourceVersion ||
+		!newEQ.GetDeletionTimestamp().IsZero() {
+		return
+	}
+
+	key, err := cache.MetaNamespaceKeyFunc(newObj)
+	if err == nil {
+		c.egressServiceQueue.Add(key)
+	}
+}
+
+// onEgressServiceDelete queues the EgressService for processing.
+func (c *Controller) onEgressServiceDelete(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %+v: %v", obj, err))
+		return
+	}
+	c.egressServiceQueue.Add(key)
+}
```

Comment:
do u need to plugin this new type with its handler to the obj retry lib ?

Reply:
yep it's level driven